lib/src/repository/traits/lock_repository.dart

```diff
@@ -64,8 +64,9 @@ class IsLocked extends State<bool> {
 }
 
 class LockTrait extends Trait {
-  LockTrait(State state, List<Property> properties)
-      : super('lock', state, properties);
+  final defaultProperties = [SupportsIsJammed(false)];
+  LockTrait(State state, [List<Property>? properties])
```

Comment:
I think it should be the other way around, 
We should be querying all possible Properties for a Trait and build this Trait class with them.

We define a default set of States and we add or remove States depending on the value of the Propert(ies).



Our graphQL query should retrieve all Properties available for a Trait. The properties themselves will determine if a State should be available or not.

Reply:
Yeah, the last line being significant in that many of the properties are around supporting states.